src/store/actions.ts

```diff
@@ -246,9 +246,9 @@ export default {
         return Promise.reject(new Error('Failed to submit order'))
       })
     }
+    dispatch('givePermissionToOrder')
```

Comment:
Why did you change the order, this is not going to work in case of a order that is being created at submit time